src/Angular.js

```diff
@@ -1772,6 +1772,48 @@ function getTestability(rootElement) {
   return injector.get('$$testability');
 }
 
+function rootElementProviderFactory(rootElement) {
+  return ['$shutdownProvider', function($shutdownProvider) {
+    $shutdownProvider.register(function() {
+      if (rootElement.dealoc) {
+        rootElement.dealoc();
+      } else {
+        rootElement.find('*').removeData();
```

Comment:
if there is an outer and an inner app, and the outer app is shutdown (remember that this will trigger a `$destroy` event on `$rootScope`), I would expect that it is reasonable for the inner app to also get destroyed.